TGC.MonoGame.Samples/Samples/Collisions/CameraCollision.cs

```diff
@@ -0,0 +1,380 @@
+            var orbitalPosition = robotBackDirection * CameraFollowRadius;
+
+
+            // We will move the Camera in the Y axis by a given distance, relative to the Robot
+            var upDistance = Vector3.Up * CameraUpDistance;
+
+            // Calculate the new Camera Position by using the Robot Position, then adding the vector orbitalPosition that sends 
+            // the camera further in the back of the Robot, and then we move it up by a given distance
+            var newCameraPosition = RobotPosition + orbitalPosition + upDistance;
+
+            // Check if the Camera collided with the scene
+            var collisionDistance = CameraCollided(newCameraPosition);
+
+            // If the Camera collided with the scene
+            if(collisionDistance.HasValue)
+            {
+                // Limit our Horizontal Radius by subtracting the distance from the collision
+                // Then clamp that value to be in between the near plane and the original Horizontal Radius
+                var clampedDistance = MathHelper.Clamp(CameraFollowRadius - collisionDistance.Value, 0.1f, CameraFollowRadius);
+
+                // Calculate our new Horizontal Position by using the Robot Back Direction multiplied by our new range
+                // (a range we know doesn't collide with the scene)
+                var recalculatedPosition = robotBackDirection * clampedDistance;
+                
+                // Set our new position. Up is unaffected
+                Camera.Position = RobotPosition + recalculatedPosition + upDistance;
+            }
+            // If the Camera didn't collide with the scene
+            else
+                Camera.Position = newCameraPosition;
+
+            // Set our Target as the Robot, the Camera needs to be always pointing to it
+            Camera.TargetPosition = RobotPosition;
+
+            // Build our View matrix from the Position and TargetPosition
+            Camera.BuildView();
+        }
+
+        /// <summary>
+        ///     Tests the Camera against the scene and checks if it collides with any wall
+        /// </summary>
+        /// <param name="cameraPosition">The Camera Position to test</param>
+        /// <returns>No value if there was no collision, the scalar distance to a wall otherwise</returns>
+        private float? CameraCollided(Vector3 cameraPosition)
+        {
+            // Create a Ray that goes from the Robot Position to the Camera
+            // The direction must be normalized
+            var difference = RobotPosition - cameraPosition;
+            var distanceToRobot = Vector3.Distance(RobotPosition, cameraPosition);
+            var normalizedDifference = difference / distanceToRobot;
+
+            Ray cameraToPlayerRay = new Ray(cameraPosition, normalizedDifference);
+
+            // Test our ray against every wall Bounding Box
+            for (int index = 0; index < WallBoxes.Length; index++)
```

Comment:
Cambiarlo por  foreach (var box in WallBoxes)?

Reply:
Ya discutimos que no esto, creo.